domain/lib/src/repository/document/document_repository.dart

```diff
@@ -60,9 +60,19 @@ abstract class DocumentRepository {
 
   Future<List<Document>> copyToMySpace(CopyRequest copyRequest);
 
-  Future<String> downloadPreviewDocument(Document document, DownloadPreviewType downloadPreviewType, Token token, Uri baseUrl, CancelToken cancelToken);
+  Future<String> downloadPreviewDocument(Document document, DownloadPreviewType downloadPreviewType, Token token, Uri baseUrl, CancelToken cancelToken, {DownloadType downloadType = DownloadType.PREVIEW});
```

Comment:
No. IMO, it not a good way. SOLID please.

Reply:
Why not?

Reply:
the purpose of this API is for `preview`. I don't want to use it for other functions. This is not follow S: `Single-responsiblity Principle`